hedera-node/src/main/java/com/hedera/services/store/tokens/HederaTokenStore.java

```diff
@@ -332,20 +341,71 @@ public ResponseCodeEnum changeOwner(NftId nftId, AccountID from, AccountID to) {
 			final var fromThisNftsOwned = (long) tokenRelsLedger.get(fromRel, TOKEN_BALANCE);
 			final var toNftsOwned = (long) accountsLedger.get(to, NUM_NFTS_OWNED);
 			final var toThisNftsOwned = (long) tokenRelsLedger.get(asTokenRel(to, nftType), TOKEN_BALANCE);
-			nftsLedger.set(nftId, OWNER, EntityId.fromGrpcAccountId(to));
+
+			if (isTreasuryForToken(to, tId)) {
+				// The default sentinel account is used (0.0.0) to represent unique tokens owned by the Treasury
+				nftsLedger.set(nftId, OWNER, EntityId.MISSING_ENTITY_ID);
+			} else {
+				nftsLedger.set(nftId, OWNER, EntityId.fromGrpcAccountId(to));
+			}
+
 			accountsLedger.set(from, NUM_NFTS_OWNED, fromNftsOwned - 1);
 			accountsLedger.set(to, NUM_NFTS_OWNED, toNftsOwned + 1);
 			tokenRelsLedger.set(fromRel, TOKEN_BALANCE, fromThisNftsOwned - 1);
 			tokenRelsLedger.set(toRel, TOKEN_BALANCE, toThisNftsOwned + 1);
 
 			var merkleUniqueTokenId = new MerkleUniqueTokenId(fromGrpcTokenId(nftId.tokenId()), nftId.serialNo());
-			this.uniqueOwnershipAssociations.get().disassociate(
-					fromGrpcAccountId(from),
-					merkleUniqueTokenId);
+			if (isTreasuryForToken(from, tId)) {
+				this.uniqueOwnershipTreasuryAssociations.get().disassociate(
+						fromGrpcTokenId(tId),
+						merkleUniqueTokenId);
+			} else {
+				this.uniqueOwnershipAssociations.get().disassociate(
+						fromGrpcAccountId(from),
+						merkleUniqueTokenId);
+			}
 
-			this.uniqueOwnershipAssociations.get().associate(
-					fromGrpcAccountId(to),
-					merkleUniqueTokenId);
+			if (isTreasuryForToken(to, tId)) {
+				this.uniqueOwnershipTreasuryAssociations.get().associate(
+						fromGrpcTokenId(tId),
+						merkleUniqueTokenId);
+			} else {
+				this.uniqueOwnershipAssociations.get().associate(
+						fromGrpcAccountId(to),
+						merkleUniqueTokenId);
+			}
+
+			hederaLedger.updateOwnershipChanges(nftId, from, to);
+
+			return OK;
+		});
+	}
+
+	@Override
+	public ResponseCodeEnum changeOwnerWildCard(NftId nftId, AccountID from, AccountID to) {
```

Comment:
Since already "knows" to transfer magic serial no `-1`, it might be clearer to just pass in the `TokenID` whose treasury is changing.

Reply:
I thought that way the code is clear to read and similar to changeOwner() with from and to. If you insist on making this change, i'll do it asap, but it will require lots of unit test refactoring and part of the business logic updated as well. What do you say? :)

Reply:
If possible i can do this in a separate PR once this once is merged, so we reduce the code base that is reviewed ;)